modules/tinymce/src/themes/silver/main/ts/backstage/UrlInputBackstage.ts

```diff
@@ -57,44 +57,34 @@ export interface UiFactoryBackstageForUrlInput {
   getUrlPicker: (filetype: string) => Option<UrlPicker>;
 }
 
-const hasOwnProperty = Object.prototype.hasOwnProperty;
-
 const isTruthy = (value: any) => !!value;
 
 const makeMap = (value: any): Record<string, boolean> => Obj.map(Tools.makeMap(value, /[, ]/), isTruthy);
 
-const getOpt = <T, K extends keyof T> (obj: T, key: K): Option<T[K]> =>
-  hasOwnProperty.call(obj, key) ? Option.some(obj[key]) : Option.none();
-
-const getTextSetting = (settings: Record<string, any>, name: string, defaultValue: string): Option<string> => {
-  const value = getOpt(settings, name).getOr(defaultValue);
-  return Type.isString(value) ? Option.some(value) : Option.none();
-};
-
-const getPicker = (settings: Record<string, any>): Option<Picker> => Option.some(settings.file_picker_callback).filter(Type.isFunction) as Option<Picker>;
+const getPicker = (editor: Editor): Option<Picker> => Option.from(editor.getParam('file_picker_callback')).filter(Type.isFunction) as Option<Picker>;
 
-const getPickerTypes = (settings: Record<string, any>): boolean | Record<string, boolean> => {
-  const optFileTypes = Option.some(settings.file_picker_types).filter(isTruthy);
-  const optLegacyTypes = Option.some(settings.file_browser_callback_types).filter(isTruthy);
+const getPickerTypes = (editor: Editor): boolean | Record<string, boolean> => {
+  const optFileTypes = Option.from(editor.getParam('file_picker_types'));
```

Comment:
These should be in `Settings` as well.

Reply:
Additionally, this changes the behaviour, as you've dropped the `isTruthy` filter.

Reply:
But instead using `from` instead of `some` which, if I've understood it correctly, will still generate an Options.none() in the case of undefined?

Reply:
No, that's incorrect. `Option.from` will only return `none` when it's undefined or null. If it's just a falsy value, such as `false` then it'll still be `some(false)`.

Reply:
Ah, then I misunderstood, my bad, good catch. I'll fix it.